cmd/codegen/generator/go/templates/modules.go

```diff
@@ -620,9 +633,34 @@ func (ps *parseState) declForFunc(fnType *types.Func) (*ast.FuncDecl, error) {
 		}
 		for _, decl := range f.Decls {
 			fnDecl, ok := decl.(*ast.FuncDecl)
-			if ok && fnDecl.Name.Name == fnType.Name() {
-				return fnDecl, nil
+			if !ok {
```

Comment:
very nit: the logic here LGTM but I do wonder if there's a simpler route by checking whether the `ast.FuncDecl` is on the same line as `fnType` (which could plausibly be checked by `fnType.Pos()`?)

If that's not actually simpler or otherwise doesn't matter in your opinion, please ignore

Reply:
Ahhhhh, I hadn't thought of this. I can't see any reason why that *wouldn't* work.

That said, meddling with the tokens/ast for the field params and comment parsing has been a *huge* pain (citation, #6279). The current approach feels a bit more complex, but also, a lot of my assumptions about the go parser and line mappings appear to keep being incorrect, so I'd rather keep it as is for now.